pkg/syncer/byokoperator/vcenter/provider.go

```diff
@@ -0,0 +1,103 @@
+package vcenter
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/vmware/govmomi/pbm"
+	cnsvsphere "sigs.k8s.io/vsphere-csi-driver/v3/pkg/common/cns-lib/vsphere"
+	cnsconfig "sigs.k8s.io/vsphere-csi-driver/v3/pkg/common/config"
+)
+
+type Provider interface {
+	DoesProfileSupportEncryption(ctx context.Context, profileID string) (bool, error)
+	ResetVC()
+}
+
+func NewProvider() Provider {
+	return &defaultProvider{}
+}
+
+type defaultProvider struct {
+	client       *cnsvsphere.VirtualCenter
+	reinitialize bool
+	lock         sync.RWMutex
+}
+
+func (p *defaultProvider) DoesProfileSupportEncryption(ctx context.Context, profileID string) (bool, error) {
```

Comment:
Can you move this function to common pbm.go file
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/vsphere/pbm.go

Reply:
It's already added in there as SupportsEncryption func. This type represents a wrapper for transparent update of vCenter credentials via ResetVC. This way, client does not need to worry about authentication.